syntax = "proto3";

package NNetwork;

message TNetwork {
    repeated TBlock block = 1;
}

message TBlock {
    optional TBlockHeader BlockHeader = 1;
    optional TBlockData BlockData = 2;
}   

message TBlockHeader {
    optional int32 BlockSize = 1;               // 4 bytes
    optional bytes PrevBlockHeaderHash = 2;     // 32 bytes (256 bits)
    optional bytes MerkleTreeHash = 3;          // 32 bytes (256 bits)
    optional float Timestamp = 4;
    optional int32 Nonce = 5;
}

message TBlockData {
    repeated bytes Transaction = 1;             // 226 bytes
}
